src/main/java/analysis/emissions/RunEmissionAnalysisByVehicleCategory.java

```diff
@@ -0,0 +1,328 @@
+        }
+
+        EventsManager eventsManager = EventsUtils.createEventsManager();
+
+        AbstractModule module = new AbstractModule() {
+            @Override
+            public void install() {
+                bind(Scenario.class).toInstance(scenario);
+                bind(EventsManager.class).toInstance(eventsManager);
+                bind(EmissionModule.class);
+            }
+        };
+
+        com.google.inject.Injector injector = Injector.createInjector(config, module);
+
+        EmissionModule emissionModule = injector.getInstance(EmissionModule.class);
+
+        EventWriterXML emissionEventWriter = new EventWriterXML(outputEmissionsFile);
+        emissionModule.getEmissionEventsManager().addHandler(emissionEventWriter);
+
+        // necessary for link emissions [g/m] output
+        LinkEmissionsHandler emissionsEventHandler = new LinkEmissionsHandler();
+        eventsManager.addHandler(emissionsEventHandler);
+
+        eventsManager.initProcessing();
+        MatsimEventsReader matsimEventsReader = new MatsimEventsReader(eventsManager);
+        matsimEventsReader.readFile(eventsFile);
+        log.info("-------------------------------------------------");
+        log.info("Done reading the events file");
+        log.info("Finish processing...");
+        eventsManager.finishProcessing();
+        log.info("Closing events file...");
+        emissionEventWriter.closeFile();
+        log.info("Done");
+        log.info("Writing (more) output...");
+
+        { // writing emissions (per link) per meter
+            File file1 = new File(linkEmissionPerMOutputFile);
+            BufferedWriter bw1 = new BufferedWriter(new FileWriter(file1));
+
+            bw1.write("linkId");
+
+            for (Pollutant pollutant : Pollutant.values()) {
+                bw1.write(";" + pollutant + " [g/m]");
+            }
+            bw1.newLine();
+
+            Map<Id<Link>, Map<Pollutant, Double>> link2pollutants = emissionsEventHandler.getLink2pollutants();
+
+            for (Id<Link> linkId : link2pollutants.keySet()) {
+                bw1.write(linkId.toString());
+
+                for (Pollutant pollutant : Pollutant.values()) {
+                    double emission = 0.;
+                    if (link2pollutants.get(linkId).get(pollutant) != null) {
+                        emission = link2pollutants.get(linkId).get(pollutant);
+                    }
+
+                    double emissionPerM = Double.NaN;
+                    Link link = scenario.getNetwork().getLinks().get(linkId);
+                    if (link != null) {
+                        emissionPerM = emission / link.getLength();
+                    }
+
+                    bw1.write(";" + emissionPerM);
+                }
+                bw1.newLine();
+            }
+
+            bw1.close();
+            log.info("Done");
+            log.info("Output written to " + linkEmissionPerMOutputFile);
```

Comment:
That is also something, that we should (I think) put into the emissions contrib, since we will need it all over the place :) 

Then we have set the format fixed for the simwrapper (or QGIS,..) analysis. When doing so, we should talk to Billy, what format is best.

@rgraebe : Is one of the part for the "general emisions calculation"-task that is on my list for you :) Nothing to do now, but please have it in mind, that we then remove it also from here (and other repos as well)

Reply:
Thank you @kt86 for your comments. Very valuable indeed. I will take a look at them first before asking for @rakow and @simei94’s review a bit later. Then at least there might be some duplicate code we can avoid by adding these generic analyses to the emissions contrib. So, until then @simei94, we can put the PR on hold while I work on and maybe ask @rakow for a few pointers to help clear some of these necessary changes.